### PR TITLE
Add macos artifact uploading

### PR DESCRIPTION
**Pull request by Pedro-Beirao**
https://github.com/Pedro-Beirao/dsda-doom/actions/runs/11133559533

Draft as Im not sure this is the best use of cpack.

I understand that cpack organizes the relevant files as if they were installed, but that isnt really what we need on macos.
Yeah we have App Bundles, but a source port that requires command line parameters on every run shouldnt be packaged that way.

___

Anyway, since we are not using an App Bundle, and the executable and dylibs are not being codesigned, the users will need to give permissions to every single dylib. So maybe we could include a txt file with a command to do it automatically (`xattr -dr com.apple.quarantine path/to/folder`)

Note: with the launcher, the users dont need to give perms like this

CC @rfomin

### CODE REVIEW

**Comment by rfomin** [comment]
I can't check it at the moment, but how does it compare to [chocolate-doom-3.1.0.dmg](https://github.com/chocolate-doom/chocolate-doom/releases/download/chocolate-doom-3.1.0/chocolate-doom-3.1.0.dmg)? Since we already have the [dsda-launcher](https://github.com/Pedro-Beirao/dsda-launcher), maybe we should bundle it somehow for the macOS release?

**Comment by Pedro-Beirao** [comment]
Chocolate has an extremely simplistic launcher for mac, so bundling it in a dmg is fine.
But, if users want to use the comand line or a different launcher, they would need to open the app bundle and move the executable and all the dylibs to a different place.

Thats what I dont like about bundling source ports as an app bundle.
I think its better to just distribute the game and dylibs in a zip like the windows release... and maybe put a link to the launcher?

This also makes it possible for the launcher to auto-update the game.

**Comment by rfomin** [comment]
> But, if users want to use the comand line or a different launcher, they would need to open the app bundle and move the executable and all the dylibs to a different place.

Oh, I didn't know that. Linux AppImage can still be used as a regular program from the command line, I thought the Mac bundles were the same.

**Comment by kraflab** [comment]
Do we need an instructions file about the permissions bundled in as well?

**Comment by Pedro-Beirao** [comment]
Yeah added a small instructions file (Quarantine.txt)
Maybe the name isnt very user friendly? But it makes sense in the context

https://github.com/Pedro-Beirao/dsda-doom/releases/tag/test2-update-script

**Comment by Pedro-Beirao** [comment]
It's all ready now

**Comment by kraflab** [comment]
Maybe `Troubleshooting.txt`?

File: .github/workflows/continuous_integration.yml [hunk 1]
```diff
@@ -16,40 +16,52 @@ jobs:
...
-            extra_options: "-G Ninja -DCMAKE_INSTALL_PREFIX=/usr"
-            package_name: "linux_gcc"
-            shell: bash
-            artifact-path: build/*.appimage
-
-          - name: "macOS Clang"
-            os: macos-latest
-            build_type: "Release"
-            extra_options: "-G Ninja"
-            shell: bash
```
Comment (rfomin) [inline .github/workflows/continuous_integration.yml:52]: https://github.com/kraflab/dsda-doom/pull/513/commits/ecc078a8fc2d3425e427dd179811035517fa47bd is not correct, do not change the `matrix.config` section.

File: .github/workflows/continuous_integration.yml [hunk 2]
```diff
@@ -109,6 +121,16 @@ jobs:
...
+          rm /usr/local/bin/2to3
+          rm /usr/local/bin/2to3-3.12
+          rm /usr/local/bin/idle3
+          rm /usr/local/bin/idle3.12
+          rm /usr/local/bin/pydoc3
+          rm /usr/local/bin/pydoc3.12
+          rm /usr/local/bin/python3
+          rm /usr/local/bin/python3-config
+          rm /usr/local/bin/python3.12
+          rm /usr/local/bin/python3.12-config
```
Comment (rfomin) [inline .github/workflows/continuous_integration.yml]: It seems `brew install --overwrite python@3.12` is simpler solution. From https://github.com/Homebrew/homebrew-core/issues/173191

File: .github/workflows/continuous_integration.yml (hunk 1 above)
Comment (Pedro-Beirao) [reply .github/workflows/continuous_integration.yml:52]: Oh my bad, I used the "Resolve Conflict" button on github. It changed more than what I thought

Comment (Pedro-Beirao) [reply .github/workflows/continuous_integration.yml:52]: Now that Im on my pc, umm whats the problem? Can't I add the `macOS x64 Clang` to `matrix.config`?

The commit you linked is just github synching stuff, and it is not really part of the PR

Comment (rfomin) [reply .github/workflows/continuous_integration.yml:52]: > Can't I add the macOS x64 Clang to matrix.config?

You can, but you also changed the syntax to earlier `{ }`. I've used `>-` in MSVC configuration which is not compatible with it.

Comment (rfomin) [reply .github/workflows/continuous_integration.yml:52]: Fix MSVC build:
```diff
diff --git a/.github/workflows/continuous_integration.yml b/.github/workflows/continuous_integration.yml
index e2cea90e6..e28727a7d 100644
--- a/.github/workflows/continuous_integration.yml
+++ b/.github/workflows/continuous_integration.yml
@@ -16,52 +16,49 @@ jobs:
       fail-fast: false
       matrix:
         config:
-          - {
-              name: "MSVC x64",
-              os: windows-latest,
-              build_type: "Release",
-              extra_options: "-A x64",
-              vcpkg_triplet: x64-windows,
-              shell: bash,
-            }
-          - {
-              name: "MSYS2 UCRT64",
-              os: windows-latest,
-              build_type: "Release",
-              extra_options: "-G Ninja",
-              package_name: "mingw_x64",
-              shell: "msys2 {0}",
-              msystem: ucrt64,
-              msys-env: mingw-w64-ucrt-x86_64,
-              artifact-path: build/*.zip,
-            }
-          - {
-              name: "Linux GCC",
-              os: ubuntu-latest,
-              build_type: "Release",
-              extra_options: "-G Ninja -DCMAKE_INSTALL_PREFIX=/usr",
-              package_name: "linux_gcc",
-              shell: bash,
-              artifact-path: build/*.appimage,
-            }
-          - {
-              name: "macOS arm64 Clang",
-              os: macos-latest,
-              build_type: "Release",
-              extra_options: "-G Ninja",
-              package_name: "mac_arm64",
-              shell: bash,
-              artifact-path: build/*.zip,
-            }
-          - {
-              name: "macOS x64 Clang",
-              os: macos-13,
-              build_type: "Release",
-              extra_options: "-G Ninja",
-              package_name: "mac_x64",
-              shell: bash,
-              artifact-path: build/*.zip,
-            }
+          - name: "MSVC x64"
+            os: windows-latest,
+            build_type: "Release"
+            vcpkg_triplet: x64-windows
+            shell: pwsh,
+            extra_options: >-
+              -A x64
+              -DCMAKE_TOOLCHAIN_FILE="${env:VCPKG_INSTALLATION_ROOT}\scripts\buildsystems\vcpkg.cmake"
+              -DVCPKG_TARGET_TRIPLET=x64-windows
+
+          - name: "MSYS2 UCRT64"
+            os: windows-latest
+            build_type: "Release"
+            extra_options: "-G Ninja"
+            package_name: "mingw_x64"
+            shell: "msys2 {0}"
+            msystem: ucrt64
+            msys-env: mingw-w64-ucrt-x86_64
+            artifact-path: build/*.zip
+
+          - name: "Linux GCC"
+            os: ubuntu-latest
+            build_type: "Release"
+            extra_options: "-G Ninja -DCMAKE_INSTALL_PREFIX=/usr"
+            package_name: "linux_gcc"
+            shell: bash
+            artifact-path: build/*.appimage
+
+          - name: "macOS arm64 Clang"
+            os: macos-latest
+            build_type: "Release"
+            extra_options: "-G Ninja"
+            package_name: "mac_arm64"
+            shell: bash
+            artifact-path: build/*.zip
+
+          - name: "macOS x64 Clang"
+            os: macos-13
+            build_type: "Release"
+            extra_options: "-G Ninja"
+            package_name: "mac_x64"
+            shell: bash
+            artifact-path: build/*.zip
 
     steps:
       - uses: actions/checkout@v4
```

Comment (Pedro-Beirao) [reply .github/workflows/continuous_integration.yml:52]: I understand now. Done in https://github.com/kraflab/dsda-doom/pull/513/commits/21105ca422f2622cea7e7f3191c3f6cd73f98d4e

Sorry about that 😅

**Comment by Pedro-Beirao** [comment]
All done... sigh

Here's also a script I made for auto-updating on mac. Should be easy adaptable for the other OSes
[dsda-update-macos.sh.zip](https://github.com/user-attachments/files/17330951/dsda-update-macos.sh.zip)

**Comment by kraflab** [comment]
Thanks 👍

**Pull request merged**